src/runtime/builtin_modules/builtins.cpp

```diff
@@ -407,11 +407,15 @@ Box* getattrFunc(Box* obj, Box* _str, Box* default_value) {
     BoxedString* str = static_cast<BoxedString*>(_str);
 
     Box* rtn = NULL;
-    try {
-        rtn = getattr(obj, str->s.data());
-    } catch (ExcInfo e) {
-        if (!e.matches(AttributeError))
-            throw e;
+    if (obj->cls == thread_local_cls) {
+        rtn = getThreadLocal(obj, const_cast<char*>(str->s.data()));
```

Comment:
I don't mind having this in here for now, but let's try to think about ways to not have to special case this...

I think one option is to change our getattr* functions to use CAPI-style exceptions.  We kind of do this by returning NULL in some (but not all) cases that the attribute doesn't exist, so it wouldn't be a huge break from what we're currently doing.

Anyway, we don't have to do that now, but it'd be nice to have a note here saying why or why not thread.local deserves special treatment here.